tests/model_registry/test_model_registry_scc.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+from typing import Self
+from simple_logger.logger import get_logger
+from _pytest.fixtures import FixtureRequest
+
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from ocp_resources.deployment import Deployment
+from ocp_resources.resource import NamespacedResource
+from utilities.constants import DscComponents
+from tests.model_registry.constants import MR_NAMESPACE, MODEL_DICT, MR_INSTANCE_NAME
+
+from kubernetes.dynamic import DynamicClient
+from ocp_utilities.infra import get_pods_by_name_prefix
+
+KEYS_TO_VALIDATE = ["runAsGroup", "runAsUser"]
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.fixture(scope="class")
+def model_registry_scc_namespace(model_registry_namespace: Namespace):
+    mr_annotations = model_registry_namespace.instance.metadata.annotations
+    return {
+        "selinuxOtions": mr_annotations.get("openshift.io/sa.scc.mcs"),
+        "uid-range": mr_annotations.get("openshift.io/sa.scc.uid-range"),
+    }
+
+
+@pytest.fixture(scope="class")
+def model_registry_resource(
+    request: FixtureRequest, admin_client: DynamicClient, model_registry_namespace: Namespace
+) -> NamespacedResource:
+    if request.param["kind"] == Deployment:
+        return Deployment(name=MR_INSTANCE_NAME, namespace=model_registry_namespace.name, ensure_exists=True)
+    elif request.param["kind"] == Pod:
+        pods = get_pods_by_name_prefix(
+            client=admin_client, pod_prefix=MR_INSTANCE_NAME, namespace=model_registry_namespace.name
+        )
+        if len(pods) != 1:
+            pytest.fail(
+                "Expected one model registry pod. Found: {[{pod.name: pod.status} for pod in pods] if pods else None}"
+            )
+        return pods[0]
+    else:
+        raise AssertionError(f"Invalid resouce requests: {request.param['kind']}")
+
+
+@pytest.mark.parametrize(
+    "model_registry_namespace, updated_dsc_component_state_scope_class, registered_model",
+    [
+        pytest.param(
+            {"namespace_name": MR_NAMESPACE},
+            {
+                "component_patch": {
+                    DscComponents.MODELREGISTRY: {
+                        "managementState": DscComponents.ManagementState.MANAGED,
+                        "registriesNamespace": MR_NAMESPACE,
+                    },
+                },
+            },
+            MODEL_DICT,
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("model_registry_namespace", "updated_dsc_component_state_scope_class", "registered_model")
+class TestModelRegistrySecurityContextValidation:
+    @pytest.mark.parametrize(
+        "model_registry_resource",
+        [
+            pytest.param({"kind": Deployment}),
+        ],
+        indirect=["model_registry_resource"],
+    )
+    def test_model_registry_deployment_security_context_validation(
+        self: Self,
+        model_registry_resource: NamespacedResource,
+        model_registry_namespace,
```

Comment:
typing is missing (tbh, i do not think we should add typing to tests)

Reply:
I will add it for now. But I too was wondering why should we have typing in tests. :)